OlinkAnalyze/R/olink_normalization_product.R

```diff
@@ -640,3 +640,80 @@ olink_normalization_qs <- function(lst_df,
 
   return(df_qq_norm)
 }
+
+
+
+#' Formatting the output of olink_normalization_product for seamless use with
+#' downstream OA functions.
+#'
+#' @author
+#' Danai G. Topouza
+#'
+#' @description
+#' Removes non-bridgeable assays. Replaces the NPX values of the non-reference
+#' project by the Median Centered or QS Normalized NPX, according to the
+#' Bridging Recommendation. Replaces OlinkID by the concatenation of the
+#' Explore HT and Explore 3072 OlinkIDs to record the OlinkIDs from both
+#' projects. Replaces SampleID with the concatenation of SampleID and Project
+#' to make unique sample IDs for downstream analysis. Removes internal and
+#' external controls. Removes BridgingRecommendation, MedianCenteredNPX,
+#'QSNormalizedNPX, OlinkID_E3072 columns.
+#'
+#' @param df A "tibble" of Olink data in long format resulting from the
+#' olink_normalization_product function.
+#'
+#' @return A "tibble" of Olink data in long format containing both input
+#' datasets with the quantile normalized quantifications, with the above
+#' modifications.
+#'
+#' @examples
+#' \donttest{
+#' # Bridge samples
+#' bridge_samples <- intersect(
+#'   x = unique(OlinkAnalyze:::data_ht_small$SampleID),
+#'   y = unique(OlinkAnalyze:::data_3k_small$SampleID)
+#' ) |>
+#'   (\(x) x[!grepl("CONTROL", x)])()
+#'
+#' # Run olink_normalization_product
+#' npx_br_data <- olink_normalization(
+#' df1 = OlinkAnalyze:::data_ht_small,
+#' df2 = OlinkAnalyze:::data_3k_small,
+#' overlapping_samples_df1 = bridge_samples,
+#' df1_project_nr = "Explore HT",
+#' df2_project_nr = "Explore 3072",
+#' reference_project = "Explore HT")
+#'
+#' # Format output
+#' npx_br_data_format <- OlinkAnalyze:::olink_normalization_product_format(
+#' npx_br_data)
+#'
+#' }
+
+olink_normalization_product_format <- function(df) {
+
+  ### Keep the data following BridgingRecommendation
+  df_format <- df |>
+    dplyr::filter(.data[["SampleType"]] == "SAMPLE") |> # Remove controls
```

Comment:
Removes external controls as appropriate. Looks good.

Reply:
Do we want to leave in SCs? Might be a more complicated conversation that Kathy, Amrita, and I have touched base on. May be added functionality for a later release.

Reply:
SCs have NA values for their QSNormalizedNPX, not sure how to include them. We would have to change the QS normalization function as well - may need to consult Kathy/Klev

Reply:
Kathy suggested that we keep SCs out for now, and potentially include them in a future release once we update the QC normalization function